File: 1009-complement-of-base-10-integer/1009-complement-of-base-10-integer.py
class Solution:
    def bitwiseComplement(self, n: int) -> int:
        if n==0:
            return 1
        r =0
        b=1
        while n>0:
            o =n&1
            f =1-o
            if f:
                r|=b
            n>>=1
            b<<=1
        return r

File: 1009-complement-of-base-10-integer/1009-complement-of-base-10-integer.cpp
class Solution {
public:
    int bitwiseComplement(int n) {
        if (n==0){
            return 1;
        }
        int m =0;
        int t=n;
        while(t>0){
            m=m<<1;
            m=m|1;
            t=t>>1;
        }
        
        return m^n;
    }
};